machine.py

```diff
@@ -0,0 +1,121 @@
+from typing import Dict, Set
+
+from coin import Coin
+from costumer import Costumer
+from drink import Drink
+from resource import Resource
+
+
+def print_coins(coins: Dict[Coin, int]):
+    for coin in coins.keys():
+        print(f'{coin.name}: {coins[coin]}')
+
+
+class NoSuchDrinkError(Exception):
+    def __init__(self, msg: str):
+        self.msg = msg
+
+
+class Machine:
+
+    def __init__(self, resources: Dict[Resource, int], coins: Dict[Coin, int], drinks: Set[Drink]):
+        self.resources = resources
```

Comment:
Should these be public?

Reply:
They should.
I didn't understand the question